Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModel.kt

```diff
@@ -1,32 +1,36 @@
 package de.rki.coronawarnapp.familytest.ui.consent
 
+import androidx.lifecycle.asFlow
 import androidx.lifecycle.asLiveData
 import dagger.assisted.Assisted
 import dagger.assisted.AssistedFactory
 import dagger.assisted.AssistedInject
 import de.rki.coronawarnapp.bugreporting.censors.family.FamilyTestCensor
 import de.rki.coronawarnapp.coronatest.qrcode.CoronaTestQRCode
+import de.rki.coronawarnapp.submission.TestRegistrationStateProcessor
 import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
 import de.rki.coronawarnapp.util.ui.SingleLiveEvent
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.first
-import kotlinx.coroutines.flow.map
+import de.rki.coronawarnapp.util.flow.combine
 
 class FamilyTestConsentViewModel @AssistedInject constructor(
     dispatcherProvider: DispatcherProvider,
     @Assisted private val coronaTestQRCode: CoronaTestQRCode,
     private val familyTestCensor: FamilyTestCensor,
+    private val registrationStateProcessor: TestRegistrationStateProcessor
 ) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
 
     val routeToScreen = SingleLiveEvent<FamilyTestConsentNavigationEvents>()
+    val registrationState = registrationStateProcessor.state.asLiveData2()
 
     private val personName = MutableStateFlow("")
 
-    val isValid = personName
-        .map { it.isNotEmpty() }
-        .asLiveData()
+    val isSubmittable = combine(personName, registrationState.asFlow()) { name, state ->
```

Comment:
Why would you convert a livedata to a flow, if registrationStateProcessor.state is already a flow?
Simple use registrationStateProcessor.state

Reply:
Changed to registrationStateProcessor.state